test/scale-infra/deploy_ks_core.yaml

```diff
@@ -0,0 +1,124 @@
+
+    - name: install golang
+      shell: |
+             curl -LO https://go.dev/dl/go1.23.2.linux-amd64.tar.gz
+             rm -rf /usr/local/go && tar -C /usr/local -xzf go1.23.2.linux-amd64.tar.gz
+             export PATH=$PATH:/usr/local/go/bin
+             go version
+                  
+    - name: install yq
+      shell: |
+             sudo wget -qO /usr/local/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64
+             sudo chmod a+x /usr/local/bin/yq
+
+    - name: install OCM CLI
+      shell: |
+             curl -L https://raw.githubusercontent.com/open-cluster-management-io/clusteradm/main/install.sh | bash
+             mkdir -p ocm
+             export INSTALL_DIR="$PWD/ocm"
+             curl -L https://raw.githubusercontent.com/open-cluster-management-io/clusteradm/main/install.sh | bash
+             export PATH=$PWD/ocm:$PATH
+        
+    - name: install KubeFlex
+      shell: |
+             sudo su <<EOF
+             bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubeflex/main/scripts/install-kubeflex.sh) --ensure-folder /usr/local/bin --strip-bin
+             EOF
+
+    - name: install Helm
+      shell: |
+             curl https://baltocdn.com/helm/signing.asc | gpg --dearmor | sudo tee /usr/share/keyrings/helm.gpg > /dev/null
+             sudo apt-get install apt-transport-https --yes
+             echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/helm.gpg] https://baltocdn.com/helm/stable/debian/ all main" | sudo tee /etc/apt/sources.list.d/helm-stable-debian.list
+             sudo apt-get update
+             sudo apt-get install helm
+
+    - name: clone the clusterloader2 repo
+      shell: |
+             rm -r $HOME/perf-tests
+             git clone -b release-1.31 https://github.com/kubernetes/perf-tests.git
+
+    - name: Register a variable, ignore errors and continue
+      ansible.builtin.command: kubectl config use-context kscore
+      register: result
+      ignore_errors: true
+
+    - name: Delete old KS deployment
+      shell: |
+             kubectl config rename-context kubernetes-admin@kubernetes kscore --ignore-not-found
+             helm delete ks-core  --ignore-not-found
+             helm -n open-cluster-management delete status-addon  --ignore-not-found
+             kubectl --context kscore delete ns open-cluster-management-hub open-cluster-management  --ignore-not-found
+             kubectl config unset contexts.wds1 --ignore-not-found
+             kubectl --context kscore delete -f https://raw.githubusercontent.com/kubestellar/kubestellar/refs/heads/release-"{{ ks_release }}"/example/kind-nginx-ingress-with-SSL-passthrough.yaml  --ignore-not-found
+      when: result is success
+
+    - name: install KubeStellar using core helm-cart
+      shell: |
+             export KUBESTELLAR_VERSION="{{ ks_release }}"
+             helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
```

Comment:
Doesn't this command need to set the values `kubeflex-operator.domain` and `kubeflex-operator.externalPort` so that the WECs can connect to the place where the Ingress controller is listening?

Is this an OpenShift cluster? If so then also the value `kubeflex-operator.isOpenShift` needs to be set, right?

Reply:
(A) I am using an ITS of type host.
(B) No, this infra is focused on plain Kubernetes cluster only.

Reply:
Wouldn't setting `kubeflex-operator.externalPort` here obviate the need to hack the wds1 context later?

Reply:
You're correct. I am setting now the flag `kubeflex-operator.externalPort` when deploying KS.

Reply:
I do not understand why using a host type of ControlPlane as ITS removes the need to set `kubeflex-operator.domain` so that code in a _WEC_ can open a connection to the ITS. What will be used as a domain name in the URL and why will code in a WEC be able to resolve it to a usable IP address?

Reply:
@MikeSpreitzer,  The URL will have the ip address of the ControlPlane master node for example: `https://192.168.1.56:xxx`. Hence, there is no longer need for domain name resolution.

Reply:
I think the answer is that this playbook does _not_ rely on KubeFlex to construct the its server reference.